tests/features/support/steps.js

```diff
@@ -4,30 +4,89 @@ const puppeteer = require('puppeteer');
 const interviewConstants = require('../../interview-constants');
 const scope = require('./scope');
 
+/* TODO:
+1. 'choice' to be any kind of choice - radio, checkbox,
+    dropdown, etc. Have to research the different DOM for each
+1. 'choice' to have a more specific way to access each item. For
+    example, for a list collect or other things that have multiple
+    things with the same text on the page.
+1. Figure out how to test allowing felxibility for coder. For example
+    there is placeholder text for the title of the form and if it's not
+    defined, placeholder text should appear (though that behavior may
+    bear discussion).
+1. Add links to resources on:
+   1. Clicks that trigger navigation need Promise.all: https://stackoverflow.com/a/60789449/14144258
+       > I ran into a scenario, where there was the classic POST-303-GET
+       and an input[type=submit] was involved. It seems that in this case, 
+       the click of the button won't resolve until after the associated form's 
+       submission and redirection, so the solution was to remove the waitForNavigation, 
+       because it was executed after the redirection and thus was timing out.
+
+       I think Promise.all is what's taking care of these situations.
+   1. Listening for `targetchanged`
+   1. Listening for responses
+   1. 
+
+Should post example of detecting page load or not on submit when there
+is a DOM change that you can detect. I suspect no request is
+being sent, but I could be wrong. Haven't yet figured out how
+to detect that.
+
+regex thoughts: https://stackoverflow.com/questions/171480/regex-grabbing-values-between-quotation-marks
+*/
+
+
 const INTERVIEW_URL = interviewConstants.INTERVIEW_URL;
 setDefaultTimeout(120 * 1000);
 
+let device_touch_map = {
+  mobile: 'tap',
+  pc: 'click',
+};
+
 // -- Puppeteer specific steps from hello_world.feature
 
-Given(/I start the interview/, async () => {
+Given(/I start the interview ?(.*)/, async (optional_device) => {
```

Comment:
Reword this to something like `I start the interview on a [phone/pc]`? That way it sounds more like natural language.

Reply:
Do you mean that the function would give more information about what it's looking for? I suspect a lot of people using this may not know how to look for that information, but maybe it'd be good to be more specific and have them there anyway. Can that be a future issue?

Note: `(pc|mobile)` may be the best bet as phone or tablet are both in that category.

Reply:
Oh, I remember now - I was also trying to be flexible in these ways:
1. Allow the person to just say 'I start the interview' and not have to specify a device.
2. All flexibility for words in between, like allowing 'a' or 'the' or 'an' ('an iphone').

Again, worth discussing whether that flexibility will add confusion.